final_eda.py

```diff
@@ -0,0 +1,204 @@
+                        props[col] = props[col].astype(np.int32)
+                    elif mn > np.iinfo(np.int64).min and mx < np.iinfo(np.int64).max:
+                        props[col] = props[col].astype(np.int64)
+
+                        # Make float datatypes 32 bit
+            else:
+                props[col] = props[col].astype(np.float32)
+
+            # Print new column type
+            # print("dtype after: ",props[col].dtype)
+            # print("******************************")
+
+    # Print final result
+    print("___MEMORY USAGE AFTER COMPLETION:___")
+    mem_usg = props.memory_usage().sum() / 1024 ** 2
+    print("Memory usage is: ", mem_usg, " MB")
+    print("This is ", 100 * mem_usg / start_mem_usg, "% of the initial size")
+    return props
+
+
+#----------------------------------------------------------------------------------
+#DATA IMPORT
+
+from google.colab import drive
+drive.mount('/content/drive')
+
+#import train dataset
+train_path = '/content/drive/MyDrive/project/train_data.csv'
+train_data = pd.read_csv(train_path, header=None)
+
+train_data= reduce_mem_usage(train_data)
+
+#import labels
+labels_path = '/content/drive/MyDrive/project/train_labels.csv'
+labels = pd.read_csv(labels_path, header=None,  names=['y'])
+
+labels= reduce_mem_usage(labels)
+
+#import test dataset
+test_path = '/content/drive/MyDrive/project/test_data.csv'
+test_data = pd.read_csv(test_path, header=None)
+
+test_data= reduce_mem_usage(test_data)
+
+X, y = train_data, labels
+X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.3, stratify = y, random_state=1)
+
+
+#----------------------------------------------------------------------------------
+#EDA
+
+
+#METRICS
+y.value_counts()
+#The distribution of the labels is very uneven. They appear in proportions of about 1:10.
+#That’s why we decided to choose f1-score.
+
+y_train = y_train['y'].apply(lambda x: 0 if x == -1 else 1)
+
+y_train.value_counts()
+
+X_train.shape
+
+y_train.shape
+
+X_train.describe()
+
+X_train.isnull().sum().sum()
+
+
+#----------------------------------------------------------------------------------
+#DISTRIBUTION
+def distribution(x_1: np.array):
+    """This function returns plot with distribution
+    :param x_1: np.array: train data after splitting
+    :return: sns.kdeplot(pd.DataFrame(x_1)[i]): plot with distribution
+    """
+  for i in [x_1.skew().idxmin(), x_1.skew().idxmax()]:
+    sns.kdeplot(pd.DataFrame(x_1)[i])
+    plt.legend(labels=['x_1.skew().idxmin()', 'x_1.skew().idxmax()'])
+
+distribution(X_train)
+
+#We can see a normal distribution.
+
+
+#----------------------------------------------------------------------------------
+#SCALING DATA
+
+#StandardScaler
+scaler = StandardScaler()
+
+X_train_scale = StandardScaler().fit_transform(X_train)
+
+#MinMaxScaler
+minmaxscaler = MinMaxScaler()
+X_train_minmaxscaler = MinMaxScaler().fit_transform(X_train)
```

Comment:
You can add a function to this two scalers

Reply:
Good point! I'll try to get them in order.